### ISSUE TITLE
Fix S4144 FP: Non-identical method false positive

### ISSUE THREAD

**Issue by nrankin18** [SonarSource/sonar-dotnet#9654]
### Description

Rule [S4144](https://rules.sonarsource.com/csharp/RSPEC-4144/) is raised when two methods have identical bodies but are not identical due to different return types.

### Repro steps

```cs
public record Foo(string A);
public record Bar(string A);

private static Foo Test1()
{
  string s = "A";
  return new(s);
}

private static Bar Test2() // S4144 is flagged here
{
  string s = "A";
  return new(s);
}
```
### Expected behavior

S4144 should not be raised since the methods are different (one returns Foo, the other Bar).

### Actual behavior

S4144 is raised.

### Known workarounds

Inhibit S4144

### Related information

* C#/VB.NET Plugins version: 8.2.1.96841
* Visual Studio version: 17.11.1
* dotnet version: 8.0
* SonarScanner for .NET version (if used): n/a
* Operating System: Windows 11

**Comment by sebastien-marichal** [comment]
Hello @nrankin18,

Thank you for reporting this issue; I confirm this to be a false positive.

The issue is due to the usage of implicit object creation `new()`.
When using explicit object creation, the issue does not raise.

I will add it to our backlog to fix it.

Have a nice day!

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1550](https://sonarsource.atlassian.net/browse/NET-1550)

[NET-1550]: https://sonarsource.atlassian.net/browse/NET-1550?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
